### DISCUSSION TITLE
All about Giscus

### DISCUSSION THREAD

**Discussion by SrGeneroso** [kwchang0831/svelte-QWER#20] [General]
[Giscus](https://github.com/giscus/giscus) is a comments system powered by Github Discussions. 

Generate your own setting: https://giscus.app/ . For more details, please consult their github repo.

If you do NOT wish to use giscus, turn it off by change enable to false .
```js
//user/config/site.ts
export const giscusConfig: Giscus.Config = {
  enable: false,
  ...
}
```

This commenting system is brilliant. I know many users will not have github accounts depending of whom makes the blog, but it's trustworthy enough to make an account if they want to comment.
It appears to be quite easy to setup manually or with the [settings generator](https://giscus.app/), btw, it will be awesomee to have one like that to create the config/site.ts

I see there is also the possibilitie to use [giscus svelte component](https://github.com/giscus/giscus-component) as well, but I don't know how much you have customize it or how well it plays, I'll look into it later, in just a glance I see you have link preferred color scheme to it in the svelte component but not given the option in the config/site file, anyway, it works good enough but I'd liek to see more comments in the svelte component and some info in the config file that points to the component.

I'll go check something else, I just wanted to start a discussion about this.

**Comment by kwchang0831** [comment]
Based on your feedback, I have add `origins` attributes[^1] to my giscus component. @ 6d3f27d3df8453649ed317a9d57b41ea1d0bc53b
It is now restrict the domains that can load giscus. Giscus will only load if the window.origin matches your site url in `user/config/site.ts`.

[^1]: https://github.com/giscus/giscus/blob/main/ADVANCED-USAGE.md
